runtime/driver/src/main/java/edu/snu/nemo/driver/ClientRPC.java

```diff
@@ -0,0 +1,147 @@
+                    final LocalAddressProvider localAddressProvider,
+                    final InjectionFuture<NemoDriver> nemoDriver,
+                    @Parameter(JobConf.ClientSideRPCServerHost.class) final String clientHost,
+                    @Parameter(JobConf.ClientSideRPCServerPort.class) final int clientPort) throws IOException {
+    this.nemoDriver = nemoDriver;
+    transport = transportFactory.newInstance(localAddressProvider.getLocalAddress(),
+        0, new SyncStage<>(new RPCEventHandler()), null, RETRY_COUNT, RETRY_TIMEOUT);
+    final SocketAddress clientAddress = new InetSocketAddress(clientHost, clientPort);
+    link = transport.open(clientAddress, ENCODER, LINK_LISTENER);
+  }
+
+  /**
+   * Shuts down the transport.
+   */
+  public void shutdown() {
+    ensureRunning();
+    try {
+      transport.close();
+    } catch (final Exception e) {
+      throw new RuntimeException(e);
+    } finally {
+      isClosed = true;
+    }
+  }
+
+  /**
+   * Write message to client.
+   * @param message message to send.
+   */
+  public void send(final ControlMessage.DriverToClientMessage message) {
+    ensureRunning();
+    link.write(message);
+  }
+
+  /**
+   * Handles message from client.
+   * @param message message to process
+   */
+  private void handleMessage(final ControlMessage.ClientToDriverMessage message) {
+    if (message.getType() == ControlMessage.ClientToDriverMessageType.LaunchDAG) {
+      nemoDriver.get().startSchedulingUserApplication(message.getLaunchDAG().getDag());
+    }
+  }
+
+  /**
+   * Provides event handler for messages from client.
+   */
+  private final class RPCEventHandler implements EventHandler<TransportEvent> {
+    @Override
+    public void onNext(final TransportEvent transportEvent) {
+      try {
+        final byte[] data = transportEvent.getData();
+        final ControlMessage.ClientToDriverMessage message = ControlMessage.ClientToDriverMessage.parseFrom(data);
+        handleMessage(message);
+      } catch (final InvalidProtocolBufferException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  /**
+   * Ensure the Transport is running.
+   */
+  private void ensureRunning() {
+    if (isClosed) {
+      throw new RuntimeException("The ClientRPC is already closed");
+    }
+  }
+
+  /**
+   * Provides encoder for {@link edu.snu.nemo.runtime.common.comm.ControlMessage.DriverToClientMessage}.
+   */
+  private static final class DriverToClientMessageEncoder implements Encoder<ControlMessage.DriverToClientMessage> {
+    @Override
+    public byte[] encode(final ControlMessage.DriverToClientMessage driverToClientMessage) {
+      return driverToClientMessage.toByteArray();
+    }
+  }
+
+  /**
+   * Provides {@link LinkListener}.
+   */
+  private static final class ClientRPCLinkListener implements LinkListener<ControlMessage.DriverToClientMessage> {
+
+    @Override
+    public void onSuccess(final ControlMessage.DriverToClientMessage driverToClientMessage) {
```

Comment:
What's the method for?

Reply:
It's empty. (Although we may add 'FINEST' logigng here.)

I had to create it because of the LinkListener interface.